### PR TITLE
Introduce the ViewModelBrick into BrickKit.

### PR DESCRIPTION
**Pull request by mdemaso**
-  This allows you to use Android DataBinding to remove the need to create new Brick Classes and simply use one (`ViewModelBrick`).
-  Allows you to tag Bricks and use those tags later to fetch from the DataManager.
-  Adds a cache for Bricks by LayoutID, allowing you to "remove based on brick type" while using `ViewModelBrick`.
-  Adds examples of using `ViewModelBrick` into `SimpleBrickFragment` and `AddRemoveBrickFragment`

### CODE REVIEW

File: BrickKit/bricks/src/androidTest/java/com/wayfair/brickkit/brick/BaseBrickTest.java [hunk 1]
```diff
@@ -208,6 +209,8 @@ protected int size() {
         assertEquals("--1--\n|111|\n11111\n|   |\n--1--", brick.toString(context, manager));
     }
 
+
```
Comment (thusson13) [inline BrickKit/bricks/src/androidTest/java/com/wayfair/brickkit/brick/BaseBrickTest.java]: rogue newlines

File: BrickKit/bricks/src/androidTest/java/com/wayfair/brickkit/brick/ViewModelBrickTest.java [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+
+    @Before
+    public void setup() {
+        context = InstrumentationRegistry.getTargetContext();
+        brickSize = mock(BrickSize.class);
+        brickPadding = mock(BrickPadding.class);
+    }
+
+    @Test
+    public void ViewModelBrick_SingleViewModel_Test() throws Throwable {
```
Comment (thusson13) [inline BrickKit/bricks/src/androidTest/java/com/wayfair/brickkit/brick/ViewModelBrickTest.java]: can these tests not throw?

File: BrickKit/bricks/src/main/java/com/wayfair/brickkit/brick/ViewModelBrick.java [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+import com.wayfair.brickkit.BrickViewHolder;
+import com.wayfair.brickkit.padding.BrickPadding;
+import com.wayfair.brickkit.size.BrickSize;
+
+/**
+ * This class is used as a Generic Brick that can take in any XML Layout and use DataBinding to
+ * insert information from a {@link ViewModel}.
+ */
+public class ViewModelBrick extends BaseBrick {
```
Comment (thusson13) [inline BrickKit/bricks/src/main/java/com/wayfair/brickkit/brick/ViewModelBrick.java]: I found it really helpful to have some other constructors like (ViewModel, int layoutId) for the cases where we just want to use the defaults

File: BrickKit/bricks/src/androidTest/java/com/wayfair/brickkit/brick/BaseBrickTest.java (hunk 1 above)
Comment (thusson13) [inline BrickKit/bricks/src/androidTest/java/com/wayfair/brickkit/brick/BaseBrickTest.java]: Rogue newlines

File: BrickKit/bricks/src/main/java/com/wayfair/brickkit/brick/ViewModelBrick.java (hunk 3 above)
Comment (mdemaso) [reply BrickKit/bricks/src/main/java/com/wayfair/brickkit/brick/ViewModelBrick.java]: I have made it follow the Builder Pattern, it only requires a layoutId, everything else is optional.

**Pull request closed without merging**